components/pkg-export-docker/src/lib.rs

```diff
@@ -139,23 +138,15 @@ impl Credentials {
                             })
                     })?;
 
-                let creds: Vec<String> = base64::decode(&token)
-                    .map_err(|e| Error::Base64DecodeError(e))
-                    .and_then(|decoded_token| {
-                        String::from_utf8(decoded_token)
-                            .map_err(|e| Error::InvalidToken(e))
-                            .and_then(|dts| Ok(dts.split(':').map(String::from).collect()))
-                    })?;
-
-                Ok(Credentials {
-                    username: creds[0].to_string(),
-                    password: creds[1].to_string(),
-                })
+                Ok(Credentials { token: token })
             }
             _ => {
                 Ok(Credentials {
-                    username: username.to_string(),
-                    password: password.to_string(),
+                    token: base64::encode(&format!(
+                        "{}:{}",
+                        username.to_string(),
+                        password.to_string()
+                    )),
```

Comment:
This is really the token?

The match on `_` makes me a bit nervous... is there a reason to use a string type here, as opposed to an Enum?

Reply:
yup that's really _really_ the token. Ain't no security like base64 security

Reply:
![tenor-111366692](https://user-images.githubusercontent.com/207178/33770411-733a763e-dbfb-11e7-9208-99915395e46e.gif)

Reply:
I guess you could accomplish this with an enum if one of the values was `default`. The idea behind this was you can have `amazon` and everything else defaulted to `docker` but maybe that's not the best UX.

Reply:
The inputs will be driven by the UI, and thus the API, right? We can have an enum that's got `Amazon` and `Docker` values, with an accompanying `FromStr` implementation that lets us parse input into the enum values. That way we offload some cognitive work to the compiler and make sure we don't leave any holes when adding new export destinations.